docs/library/ubluetooth.rst

```diff
@@ -185,7 +185,15 @@ Observer Role (Scanner)
     interval and window are 1.28 seconds and 11.25 milliseconds respectively
     (background scanning).
 
-    For each scan result, the ``_IRQ_SCAN_RESULT`` event will be raised.
+    For each scan result the ``_IRQ_SCAN_RESULT`` event will be raised, with event
+    data ``(addr_type, addr, adv_type, rssi, adv_data)``.  ``adv_type`` values correspond
+    to the Bluetooth Specification:
+
+        * 0x00 - ADV_IND - connectable and scannable undirected advertising
+        * 0x01 - ADV_DIRECT_IND - connectable directed advertising
+        * 0x02 - ADV_SCAN_IND - scannable undirected advertising
+        * 0x03 - ADV_NONCONN_IND - non-connectable undirected advertising
+        * 0x04 - SCAN_RSP - scan response
```

Comment:
Thanks for fixing this.

Just a question on scan response. Why does the btstack implementation have the `if type < 4` check, but NimBLE passes them all through?

Also I guess now that scan responses are supported, this means we should also enable active scans (iirc both stacks currently only do passive scans). Probably should be be off-by-default otherwise they'll now see SCAN_RESULT events for both the advertising payload and the scan response payload (which is probably a more serious breaking change than this PR).

Reply:
>  Why does the btstack implementation have the `if type < 4` check, but NimBLE passes them all through?

I originally removed that check, but testing (using `multi_bluetooth/ble_gap_advertise.py`) showed that BTstack passes through a SCAN_RSP result even for passive scanning (with an empty `adv_data` entry).  I'm not really sure what is going on with that, maybe it's not configuring passive scanning correctly, or maybe BTstack always passes SCAN_RSP through regardless, or maybe it really is doing an active scan.  Something to investigate, and maybe should do that before merging this PR.

> Also I guess now that scan responses are supported, this means we should also enable active scans.

Yes that would be good, I guess it'd be another optional argument to `gap_scan()` (maybe an integer "mode" rather than a boolean "active"?).

Reply:
Looking into the BTstack issue it's because the scan parameters are not being set correctly and it defaults to an active scan.  It could be a problem with BTstack, or with the way our code initialises BTstack; I filled a report here: https://github.com/bluekitchen/btstack/issues/275